cmd/bootstrap/cmd/key.go

```diff
@@ -9,78 +9,120 @@ import (
-	flagRole        string
-	flagAddress     string
+	flagRole    string
+	flagAddress string
+
+	// seed flags
 	flagNetworkSeed []byte
 	flagStakingSeed []byte
+	flagMachineSeed []byte
 )
 
+// keyCmd represents the key command
+var keyCmd = &cobra.Command{
+	Use:   "key",
+	Short: "Generate networking, staking, and machine account keys for a partner node and write them to files. If no key-type ('--networking', '--staking', '--machine') flags are specified, all keys are generated. If any key-type flags are specified, only the specified key types are generated. ",
+	Run:   keyCmdRun,
+}
+
+func init() {
+	rootCmd.AddCommand(keyCmd)
+
+	// required flags
+	keyCmd.Flags().StringVar(&flagRole, "role", "", "node role (can be \"collection\", \"consensus\", \"execution\", \"verification\" or \"access\")")
+	_ = keyCmd.MarkFlagRequired("role")
+	keyCmd.Flags().StringVar(&flagAddress, "address", "", "network address")
+	_ = keyCmd.MarkFlagRequired("address")
+
+	keyCmd.Flags().BytesHexVar(&flagNetworkSeed, "networking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded networking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagStakingSeed, "staking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded staking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagMachineSeed, "machine-seed", generateRandomSeed(), fmt.Sprintf("hex encoded machine account seed (min %v bytes)", minSeedBytes))
+}
+
 // keyCmdRun generate the node staking key, networking key and node information
 func keyCmdRun(_ *cobra.Command, _ []string) {
+
+	// TODO: We need NodeID of the node to continue checking if the node-info.priv.json exists
+	// TODO: Add paths to log messages for clarity
+	exists, err := pathExists(fmt.Sprintf(model.PathNodeInfoPriv, flow.Identifier{}))
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not check if node-info.priv.json exists")
+	}
+	if exists {
+		log.Warn().Msg("node-info.priv.json exists, please delete and try again if you are tying to generate new keys")
+		return
+	}
+
 	// validate inputs
 	role := validateRole(flagRole)
 	validateAddressFormat(flagAddress)
-	networkSeed := validateSeed(flagNetworkSeed)
-	stakingSeed := validateSeed(flagStakingSeed)
 
-	log.Debug().Msg("will generate networking key")
-	networkKeys, err := run.GenerateNetworkingKeys(1, [][]byte{networkSeed})
+	// generate all three keys
+	networkKey, stakingKey, machineKey, err := generateKeys()
 	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate networking key")
+		log.Fatal().Err(err).Msg("could not generate keys")
 	}
-	log.Info().Msg("generated networking key")
 
-	log.Debug().Msg("will generate staking key")
-	stakingKeys, err := run.GenerateStakingKeys(1, [][]byte{stakingSeed})
-	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate staking key")
-	}
-	log.Info().Msg("generated staking key")
+	// construct NodeMachineAccountInfo struct to write
+	machineAccountInfo := assembleNodeMachineAccountInfo(machineKey)
```

Comment:
One of the safety goals for this tool is to never overwrite an existing file that contains sensitive information. 

Here we are writing a partial `NodeMachineAccountInfo` file and writing it to the default path `PathNodeMachineAccountInfoPriv`. In the subsequent command to generate the full `NodeMachineAccountInfo` we would need to overwrite this file because it is already in the path of the final file. 

I suggest we create a separate type and canonical path for this intermediary `NodeMachineAccountPrivateKey` file so it is clear that is different from the full `NodeMachineAccountInfo` and so that we don't need to overwrite that file.